### PR TITLE
all e2e tests use the same command to run in ci

### PR DESCRIPTION
**Pull request by David-Losinski**
### Ticket:

- [18848](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/18848)

### Description

Updates commands for running e2e smoke and regression in CI to be the same for all apps (mirrored in the test pipelines in the pipeline template repo)
Adds placeholder echo statement in npm commands where the project currently has no e2e tests
Where tests are run an results.xml (junit format) is generated to be read into ADO for test reporting

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [ ] **New feature**
- [x] **Dev change**
- [ ] **Additional tests**
- [x] **Documentation**
- [ ] **Other**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

**Comment by alfonsograziano** [comment]
@David-Losinski I see you're not changing the `payments` app to include the same commands, is there any specific reason? Can we help somehow?

**Comment by David-Losinski** [comment]
> @David-Losinski I see you're not changing the `payments` app to include the same commands, is there any specific reason? Can we help somehow?

Payments was done in the PR yesterday, all others where changed to match it

**Comment by alfonsograziano** [comment]
Makes sense, thanks for the clarification :)

File: apps/messaging-api/.gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 node_modules
 .env
 .tap
+
+e2e/test-results/
```
Comment (peschina) [inline apps/messaging-api/.gitignore:5]: Can you please add a newline at EOF?

File: documentation/docs/testing.md [hunk 2]
```diff
@@ -37,4 +37,17 @@ For API e2e testing we use [Bruno](https://www.usebruno.com/). It has a Postman
...
\ No newline at end of file
+For back end performance testing we are looking at [K6](https://k6.io/)
+
+## Testing in CICD
+
+We are starting to add testing to our pipelines, as not all apps have e2e or unit tests we have added an echo comment to the package.json file so we can use a common pipeline for all projects without pipelines failing because the command does not exist for the app. Initially we have UI tests for payments and at least 1 test for most API projects (excluding upload and mock api).
+
+Current commands used and mapped to the shared building block pipelines:
+"test:smoke:e2e" - used for smoke tests on depolyment to develop (staging comming soon) 
+"test:regression:e2e" - runs all tests overnight against develop (comming soon)
```
Comment (peschina) [inline documentation/docs/testing.md:48]:
```suggestion
"test:smoke:e2e" - used for smoke tests on deployment to dev (staging coming soon) 
"test:regression:e2e" - runs all tests overnight against dev (coming soon)
```

**Review by peschina — DISMISSED** [review]
LGTM, a few minor comments

File: apps/messaging-api/.gitignore (hunk 1 above)
Comment (David-Losinski) [reply apps/messaging-api/.gitignore:5]: done

**Review by Thurfjell — APPROVED** [review]

**Pull request merged**